pallets/utxo/src/header.rs

```diff
@@ -59,142 +213,120 @@ u16_to_enum! {
     }
 }
 
-impl SignatureMethod {
-    pub fn extract(header: TXOutputHeader) -> Result<SignatureMethod, &'static str> {
-        SignatureMethod::try_from(header & 7u16)
-    }
-
-    pub(crate) fn insert(header: &mut TXOutputHeader, signature_method: SignatureMethod) {
-        *header = header.clone() & 0b1_111111_111111_000; // remove the original signature, if any.
-        let signature_method = signature_method as u16;
-        *header = header.clone() | signature_method;
+u128_to_enum! {
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Clone, Encode, Decode, Eq, PartialEq, PartialOrd, Ord, Hash, Debug)]
+pub enum TokenType {
+    MLT = 0,
```

Comment:
MLT is technically not a MLS-01 token so does it need a token ID? If we remove it how can we distinguish MLS-01 and MLT?

Reply:
In your case, you can check `TokenType`. Now `TokenID` and `TokenType` are different things.